applications/nrf_desktop/src/modules/dfu.c

```diff
@@ -558,6 +565,14 @@ static void handle_dfu_start(const uint8_t *data, const size_t size)
 
 		terminate_dfu();
 	} else {
+		flash_write_block = flash_area_align(flash_area);
```

Comment:
Also, could we perform the assignments and do the checks on boot? So that user would instantly be notified about invalid configuration